src/mock_crypto.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// http://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Mock cryptographic primitives.
+//!
+//! These primitives are designed to be very fast (several times faster than the real ones), but
+//! they are NOT secure. They are supposed to be used for testing only.
+
+/// Mock version of a subset of the `rust_sodium` crate.
+pub mod rust_sodium {
```

Comment:
Making every `pub` in this file `pub(crate)` would make the scope of these clearer and help avoid any of them being accidentally exported in the future?  (We'd need an exception for the two init functions if they are to be exported).